### DISCUSSION TITLE
Can a node which is not a leader initiate a write without rerouting?

### DISCUSSION THREAD

**Discussion by natilivni** [dotnet/dotNext#32] [Q&A] [answered]
Let's say I have an endpoint which needs to do a write operation to the audit log, the do a bunch of other things which are better done distributed rather than on the leader. Is there a way to:

1. Initiate a write to the audit log on the leader from my follower node. 
2. Wait until the write is a both commited to the audit log as well as well as replicated to my follower node.
3. Continue with app logic on follower node.

**Comment by sakno** [comment, marked as answer]
Of course. Any follower node is able to send write command to the leader node. In case of ASP.NET Core application, you can redirect client of your cluster directly to the leader node. Read [this](https://sakno.github.io/dotNext/features/cluster/aspnetcore.html#redirection-to-leader) article for more information.

If you want to perform write on the follower node then use `IMessageBus.LeaderRouter` property to send a message to the leader node. [IMessageBus](https://sakno.github.io/dotNext/api/DotNext.Net.Cluster.Messaging.IMessageBus.html) interface is implemented by Raft cluster. Take a look at [this](https://sakno.github.io/dotNext/features/cluster/index.html) article which describe cluster programming in details.
